lib/inc/drogon/HttpFilter.h

```diff
@@ -60,8 +59,34 @@ class HttpFilter : public DrObject<T>, public HttpFilterBase
 {
   public:
     static constexpr bool isAutoCreation{AutoCreation};
-    virtual ~HttpFilter()
+    ~HttpFilter() override = default;
+};
+
+template <typename T, bool AutoCreation = true>
+class HttpCoroFilter : public DrObject<T>, public HttpFilterBase
+{
+  public:
+    static constexpr bool isAutoCreation{AutoCreation};
+    ~HttpCoroFilter() override = default;
+    void doFilter(const HttpRequestPtr &req,
+                  FilterCallback &&fcb,
+                  FilterChainCallback &&fccb) final
     {
+        [this](const HttpRequestPtr &req,
+               FilterCallback &&fcb,
+               FilterChainCallback &&fccb) -> AsyncTask {
```

Comment:
I think the `async_run` method should be used here to swap the temporary lambda object.

As follows:
https://github.com/drogonframework/drogon/blob/4ef31d7c2d6a76362caf9d34a5f55d4edbce9cb2/examples/redis_cache/filters/TimeFilter.cc#L13-L26